addons/binding/org.openhab.binding.geofence/src/main/java/org/openhab/binding/geofence/internal/BindingComponent.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.geofence.internal;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.i18n.LocationProvider;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.geofence.internal.config.BindingConfiguration;
+import org.openhab.binding.geofence.internal.discovery.DeviceDiscoveryService;
+import org.openhab.binding.geofence.internal.handler.TrackerDeviceHandler;
+import org.openhab.binding.geofence.internal.message.NotificationBroker;
+import org.openhab.binding.geofence.internal.provider.gpslogger.GPSLoggerCallbackServlet;
+import org.openhab.binding.geofence.internal.provider.owntracks.OwnTracksCallbackServlet;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ManagedService;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.openhab.binding.geofence.internal.BindingConstants.CONFIG_PID;
+
+/**
+ * Main component
+ *
+ * @author Gabor Bicskei - Initial contribution
+ */
+@Component(
+        configurationPid = CONFIG_PID,
+        immediate = true,
+        service = {ThingHandlerFactory.class, ManagedService.class}
```

Comment:
You must not implement `ManagedService`. Instead, the component itself has a configuration, which is passed through the `activate` method. See e.g. the [Sonos binding](https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosHandlerFactory.java#L67-L72) for a reference.